lib/dialects/abstract/query.js

```diff
@@ -514,7 +514,19 @@ AbstractQuery.prototype.isUpdateQuery = function() {
 
 AbstractQuery.prototype.handleSelectQuery = function(results) {
   var result = null;
-
+  // Map raw fields to arbitrary names if a mapping is provided
+  if (this.options.fieldMap) {
```

Comment:
Currently the fix also works for postgres `returning: true` on `update`. We'd have to put it behind  `this.options.raw || this.options.type === QueryTypes.BULKUPDATE`.